internal/node/node_loader.js

```diff
@@ -83,20 +83,24 @@ function resolveToModuleRoot(path) {
  * See https://github.com/bazelbuild/bazel/issues/3726
  */
 function loadRunfilesManifest(manifestPath) {
-  const result = Object.create(null);
+  const runfilesManifest = Object.create(null);
+  const reverseRunfilesManifest = Object.create(null);
   const input = fs.readFileSync(manifestPath, {encoding: 'utf-8'});
   for (const line of input.split('\n')) {
     if (!line) continue;
     const [runfilesPath, realPath] = line.split(' ');
-    result[runfilesPath] = realPath;
+    runfilesManifest[runfilesPath] = realPath;
+    reverseRunfilesManifest[realPath] = runfilesPath;
   }
-  return result;
+  return {runfilesManifest, reverseRunfilesManifest};
 }
-const runfilesManifest =
+const { runfilesManifest, reverseRunfilesManifest } =
     // On Windows, Bazel sets RUNFILES_MANIFEST_ONLY=1.
     // On every platform, Bazel also sets RUNFILES_MANIFEST_FILE, but on Linux
     // and macOS it's faster to use the symlinks in RUNFILES_DIR rather than resolve
-    // through the indirection of the manifest file
+    // through the indirection of the manifest file.
+    // We also need to construct a reverse map to make sure to always resolve files from
```

Comment:
this doesn't seem to match the implementation below - you don't add a new place where we throw or reject input right? We don't need any guarantee that we only ever resolve declared inputs, that should be bazel's job (like with the docker sandboxing). We only need happy path cases, I believe

Reply:
Yeah this comment was old, back when I was trying a different approach. Have updated it to the current implementation.